flips/20220811-fungible-tokens-metadata.md

```diff
@@ -0,0 +1,210 @@
+        pub let publicLinkedType: Type
+
+        /// Type that should be linked at the aforementioned private path. This 
+        /// is normally a restricted type with at a minimum the `FT.Provider` 
+        /// interface
+        pub let providerLinkedType: Type
+
+        /// Function that allows creation of an empty FT vault that is intended
+        /// to store the funds.
+        pub let createEmptyVault: ((): @FungibleToken.Vault)
+
+        init(
+            storagePath: StoragePath,
+            publicPath: PublicPath,
+            providerPath: PrivatePath,
+            publicCollection: Type,
+            publicLinkedType: Type,
+            providerLinkedType: Type,
+            createEmptyVaultFunction: ((): @FungibleToken.Vault)
+        ) {
+            pre {
+                publicLinkedType.isSubtype(of: Type<&{FungibleToken.Receiver, FungibleToken.Balance, Resolver}>()): "Public type must include FungibleToken.Receiver, FungibleToken.Balance and FungibleMetadataViews.Resolver interfaces."
+                providerLinkedType.isSubtype(of: Type<&{FungibleToken.Provider, Resolver}>()): "Provider type must include NonFungibleToken.Provider and FungibleMetadataViews.Resolver interface."
+            }
+            self.storagePath=storagePath
+            self.publicPath=publicPath
+            self.providerPath = providerPath
+            self.publicLinkedType=publicLinkedType
+            self.providerLinkedType = providerLinkedType
+            self.createEmptyVault = createEmptyVaultFunction
+        }
+    }
+```
+
+In order to be able to use the `FTDisplay` we will need to re-use or re-implement (depending on if this ends up being part of the original `MetadataViews.cdc` contract or having its own contract) the following views: `ExternalURL`, `File`, `HTTPFile`, `IPFSFile`, `Media` and `Medias`. 
+
+### Drawbacks
+
+Main drawback could came from inefficient implementations of views from fungible token contracts.
+
+### Alternatives Considered
+
+Since the `MetadataViews` contract has been a successful solution for increasing NFT interoperability no alternatives have been considered for the FT. 
+
+### Performance Implications
+
+Due to the potentially dynamic nature of this proposal, the methods that return metadata views can be implemented in any way that the author sees fit which could lead to some non-performant implementations.
+
+### Dependencies
+
+This proposal builds on the existing FT interface defined [here](https://github.com/onflow/flow-ft).
+
+### Engineering Impact
+
+Adhering to this standard will require contract authors to implement a few new methods as part of their FT. They will also likely need to import standard types into their contracts.
+
+There may also need to be a committee, process, and code repository put in place for managing the standard metadata types.
+
+### Best Practices
+
+Fungible Token issuers should implement the `FTView` to achieve the as much ecosystem compatibility as possible.
```

Comment:
See comment above and clarify if the two sub-views should be also provided.

Reply:
and what about the two view fields inside the main view being optional types?

```   
pub struct FTView {
        pub let ftDisplay: FTDisplay?
        pub let vaultData: FTVaultData?
        init(
            ftDisplay : FTDisplay?,
            vaultData : FTVaultData?
        ) {
            self.ftDisplay = ftDisplay
            self.vaultData = vaultData
        }
}
```

do you think this should be something that must be settled before getting the flip into the forum or that is worth of being discussed there?